server/datastore/mysql/apple_mdm.go

```diff
@@ -1950,6 +1950,79 @@ func bulkDeleteHostDiskEncryptionKeysDB(ctx context.Context, tx sqlx.ExtContext,
 	return err
 }
 
+func (ds *Datastore) MDMAppleGetEULAMetadata(ctx context.Context) (*fleet.MDMAppleEULA, error) {
+	// Currently, there can only be one EULA in the database, so we pick
+	// the first entry we find.
+	stmt := "SELECT name, created_at, token FROM eulas LIMIT 1"
+	var eula fleet.MDMAppleEULA
+	if err := sqlx.GetContext(ctx, ds.reader, &eula, stmt); err != nil {
+		if err == sql.ErrNoRows {
+			return nil, ctxerr.Wrap(ctx, notFound("MDMAppleEULA"))
+		}
+		return nil, ctxerr.Wrap(ctx, err, "get EULA metadata")
+	}
+	return &eula, nil
+}
+
+func (ds *Datastore) MDMAppleGetEULABytes(ctx context.Context, token string) (*fleet.MDMAppleEULA, error) {
+	stmt := "SELECT name, bytes FROM eulas WHERE token = ?"
+	var eula fleet.MDMAppleEULA
+	if err := sqlx.GetContext(ctx, ds.reader, &eula, stmt, token); err != nil {
+		if err == sql.ErrNoRows {
+			return nil, ctxerr.Wrap(ctx, notFound("MDMAppleEULA"))
+		}
+		return nil, ctxerr.Wrap(ctx, err, "get EULA bytes")
+	}
+	return &eula, nil
+}
+
+func (ds *Datastore) MDMAppleInsertEULA(ctx context.Context, eula *fleet.MDMAppleEULA) error {
+	// check if we already have an EULA in the database. This is a bit odd,
+	// at the moment we only allow one global EULA, but I haven't find a
+	// good way to enforce this check at the database level.
```

Comment:
In the future, are we expecting to have something like the team id / no team concept for these where there could be  at most one no team EULA and one EULA per team?